scarb/tests/build.rs

```diff
@@ -687,3 +687,58 @@ fn edition_must_exist() {
                  unknown variant `2021`, expected one of `2023_01`, `2023_10`, `2023_11`
         "#});
 }
+
+#[test]
+fn dev_dep_used_outside_tests() {
+    let t = TempDir::new().unwrap();
+    let q = t.child("q");
+    ProjectBuilder::start()
+        .name("q")
+        .lib_cairo(
+            r#"
+            fn dev_dep_function() -> felt252 {
+                42
+            }
+            "#,
+        )
+        .build(&q);
+    ProjectBuilder::start()
+        .name("x")
+        .dev_dep("q", &q)
+        .src(
+            "src/foo.cairo",
+            r#"
+        use q::dev_dep_function;
+
+        fn foo_not_working() {
+            dev_dep_function();
+        }
+        "#,
+        )
+        .lib_cairo(
+            r#"
+        mod foo;
+
+        fn not_working() {
+            foo::foo_not_working();
+        }
+        "#,
+        )
+        .build(&t);
```

Comment:
Nit: the second file does not really add a lot here. You can simplify with 
```suggestion
        .dev_dep("q", &q)
        .lib_cairo(indoc! {r#"
            use q::dev_dep_function;

            fn foo_not_working() {
                dev_dep_function();
            }
        "#})
        .build(&t);
```